services/datamanager/builtin/builtin_test.go

```diff
@@ -957,3 +959,161 @@ func getTestModelManagerConstructor(t *testing.T, server rpc.Server, zipFileName
 		return model.NewManager(logger, cfg.Cloud.ID, client, conn, &mockClient{zipFileName: zipFileName})
 	}
 }
+
+func TestDataCapture(t *testing.T) {
+	tests := []struct {
+		name                 string
+		initialDisableStatus bool
+		newDisableStatus     bool
+	}{
+		{
+			"Config with data capture disabled should capture nothing.",
+			true,
+			true,
+		},
+		{
+			"Config with data capture enabled should capture data.",
+			false,
+			false,
+		},
+		{
+			"Disabling data capture should cause it to stop.",
+			false,
+			true,
+		},
+		{
+			"Enabling data capture should cause it to begin.",
+			true,
+			false,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			tmpDir, err := ioutil.TempDir("", "")
+			defer os.RemoveAll(tmpDir)
+			test.That(t, err, test.ShouldBeNil)
+
+			rpcServer, _ := buildAndStartLocalServer(t)
+			defer func() {
+				err := rpcServer.Stop()
+				test.That(t, err, test.ShouldBeNil)
+			}()
+			dmsvc := newTestDataManager(t, "arm1", "")
+			dmsvc.SetSyncerConstructor(getTestSyncerConstructor(t, rpcServer))
+			testCfg := setupConfig(t, configPath)
+			svcConfig, ok, err := getServiceConfig(testCfg)
+			test.That(t, err, test.ShouldBeNil)
+			test.That(t, ok, test.ShouldBeTrue)
+			svcConfig.CaptureDisabled = tc.initialDisableStatus
+			svcConfig.ScheduledSyncDisabled = false
+			svcConfig.CaptureDir = tmpDir
+			err = dmsvc.Update(context.Background(), testCfg)
+			test.That(t, err, test.ShouldBeNil)
+
+			// let run for a second
+			time.Sleep(captureWaitTime)
+
+			// Check if data has been captured (or not) as we'd expect.
+			initialCaptureFiles := getAllFiles(tmpDir)
+			initialCaptureFilesSize := getTotalFileSize(initialCaptureFiles)
+			if !tc.initialDisableStatus {
+				// TODO: check contents
+				test.That(t, len(initialCaptureFiles), test.ShouldBeGreaterThan, 0)
+				test.That(t, initialCaptureFilesSize, test.ShouldBeGreaterThan, 0)
+			} else {
+				test.That(t, len(initialCaptureFiles), test.ShouldEqual, 0)
+				test.That(t, initialCaptureFilesSize, test.ShouldEqual, 0)
+			}
+
+			// change status
+			svcConfig.CaptureDisabled = tc.newDisableStatus
+			err = dmsvc.Update(context.Background(), testCfg)
+			test.That(t, err, test.ShouldBeNil)
+
+			// let run for a second
+			time.Sleep(captureWaitTime)
+			midCaptureFiles := getAllFiles(tmpDir)
+			midCaptureFilesSize := getTotalFileSize(midCaptureFiles)
+			// Let run for a second, if still on, more data should have been captured, flush data here.
+			time.Sleep(captureWaitTime)
+			updatedCaptureFiles := getAllFiles(tmpDir)
+			updatedCaptureFilesSize := getTotalFileSize(updatedCaptureFiles)
+			if !tc.newDisableStatus && tc.initialDisableStatus {
```

Comment:
Can you add comments to each of these cases saying what they are? e.g. `If disabled then enabled`

The whole disabled instead of enabled thing made it hard for me to figure it out